kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/zookeeper/ZookeeperClientProvider.scala

```diff
@@ -144,4 +144,21 @@ object ZookeeperClientProvider extends Logging {
     }
   }
 
+  /**
+   * Creates a zookeeper client before calling `f` and close it after calling `f`.
+   */
+  def withZkClient[T](conf: KyuubiConf)(f: CuratorFramework => T): T = {
+    val zkClient = buildZookeeperClient(conf)
+    try {
+      zkClient.start()
+      f(zkClient)
+    } finally {
+      try {
+        zkClient.close()
+      } catch {
+        case e: IOException => error("Failed to release the zkClient", e)
+      }
+    }
+  }
```

Comment:
seems not needed, similar with DiscoveryClientProvider::withDiscoveryClient

```
object DiscoveryClientProvider extends Logging {

  /**
   * Creates a zookeeper client before calling `f` and close it after calling `f`.
   */
  def withDiscoveryClient[T](conf: KyuubiConf)(f: DiscoveryClient => T): T = {
    val discoveryClient = createDiscoveryClient(conf)
    try {
      discoveryClient.createClient()
      f(discoveryClient)
    } finally {
      try {
        discoveryClient.closeClient()
      } catch {
        case e: IOException => error("Failed to release the zkClient", e)
      }
    }
  }
```

Reply:
![image](https://user-images.githubusercontent.com/35604105/196033979-152e8c47-5de5-4749-be53-074a6a869b85.png)   Since the creation of DistributedAtomicInteger requires a client of CuratorFramework, the DiscoveryClient provided by DiscoveryClientProvider is converted into CuratorFramework